node/service/src/lib.rs

```diff
@@ -1648,7 +1678,115 @@ mod tests {
 			announce_block: true,
 			base_path: BasePath::new(Path::new("")),
 			informant_output_format: Default::default(),
+			wasmtime_precompiled: None,
 			runtime_cache_size: 2,
 		}
 	}
 }
+
+struct DigestProvider<RuntimeApi, Executor>
+where
+	RuntimeApi: Send + Sync,
+	Executor: ExecutorT + 'static,
+{
+	client: Arc<FullClient<RuntimeApi, Executor>>,
+	keystore: Arc<dyn Keystore>,
+}
+
+impl<RuntimeApi, Executor> DigestsProvider<NimbusId, H256> for DigestProvider<RuntimeApi, Executor>
```

Comment:
`DigestsProvider` vs `DigestProvider` is a bit confusing, is this because it returns a single Digest?

Reply:
Renamed to `VrfDigestsProvider` 👍